plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,230 @@
+        try {
+            metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName())
+                    .ifPresent(table -> {
+                        Table updatedTable = setTableMetadata(table, info.version, info.schemaString, info.tableComment);
+                        metastore.replaceTable(table.getDatabaseName(), table.getTableName(), updatedTable, buildInitialPrivilegeSet(table.getOwner().orElseThrow()));
+                        log.debug("Replaced table: '%s'", schemaTableName);
+                    });
+        }
+        catch (Exception e) {
+            log.warn(e, "Failed to store table metadata for '%s'", schemaTableName);
+        }
+    }
+
+    @VisibleForTesting
+    public void clear()
+    {
+        updateInfos.clear();
+    }
+
+    @PreDestroy
+    public void stop()
+    {
+        if (enabled) {
+            scheduler.shutdownNow();
+            executor.shutdownNow();
+        }
+    }
+
+    public boolean isSameTransactionVersion(Table table, TableSnapshot snapshot)
+    {
+        if (!containsLastTransactionVersion(table)) {
+            return false;
+        }
+        return getLastTransactionVersion(table) == snapshot.getVersion();
+    }
+
+    public boolean containsLastTransactionVersion(Table table)
+    {
+        return table.getParameters().containsKey(TRINO_LAST_TRANSACTION_VERSION);
+    }
+
+    public long getLastTransactionVersion(Table table)
+    {
+        return Long.parseLong(table.getParameters().get(TRINO_LAST_TRANSACTION_VERSION));
+    }
+
+    public boolean containsSchemaString(Table table)
+    {
+        return table.getParameters().containsKey(TRINO_METADATA_SCHEMA_STRING);
+    }
+
+    public List<ColumnMetadata> getColumnsMetadata(Table table)
+    {
+        String schemaString = table.getParameters().get(TRINO_METADATA_SCHEMA_STRING);
+        // Specify NONE because physical names are unused when listing columns
+        return getColumnMetadata(schemaString, typeManager, NONE).stream()
+                .map(DeltaLakeColumnMetadata::columnMetadata)
+                .collect(toImmutableList());
+    }
+
+    public boolean canStoreTableMetadata(ConnectorSession session, String schemaString, Optional<String> tableComment)
+    {
+        return storeTableMetadata(session) &&
+                schemaString.length() <= tableParameterLengthLimit &&
+                tableComment.orElse("").length() <= tableParameterLengthLimit;
+    }
+
+    private static Table setTableMetadata(Table table, long version, String schemaString, Optional<String> tableComment)
+    {
+        ImmutableMap.Builder<String, String> parameters = ImmutableMap.builder();
+        parameters.putAll(table.getParameters());
+        setTableMetadata(parameters, version, schemaString, tableComment);
+        return Table.builder(table)
+                .setParameters(parameters.buildKeepingLast())
+                .build();
+    }
+
+    public static void setTableMetadata(
```

Comment:
inline?

Reply:
This method is used from two places (`DeltaLakeMetadata#deltaTableProperties` and another `DeltaLakeTableMetadataScheduler#setTableMetadata`).